### PR TITLE
[react-twitter-embed] Add definitions

### PR DESCRIPTION
**Pull request by zozoens31**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---
This change is [Reviewable](https://reviewable.io/reviews/definitelytyped/definitelytyped/41758)

### CODE REVIEW

File: types/react-twitter-embed/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// Minimum TypeScript Version: 3.5
+import * as React from 'react';
+
+export interface BaseProps {
+    onLoad?: (element: HTMLElement) => void;
+    options?: {};
+    placeholder?: string | React.ReactElement;
+}
+
+export namespace timeline {
```
Comment (uniqueiniquity) [inline types/react-twitter-embed/index.d.ts]: Is there a particular reason you need to wrap these types in a namespace?
As far as I can tell, it's not really helping you in any way, and you don't seem to be using it in your test file.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Review by zozoens31 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/definitelytyped/definitelytyped/41758)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @uniqueiniquity)

___
*[types/react-twitter-embed/index.d.ts, line 14 at r1](https://reviewable.io/reviews/definitelytyped/definitelytyped/41758#-LzBhhkICi_e7_-nWXjc-r1-14:-LzBhuhqA3Syv8U4H-68:b-w5hqaz) ([raw file](https://github.com/definitelytyped/definitelytyped/blob/283c90709ccd625e75a57fbd72acbc838496dadd/types/react-twitter-embed/index.d.ts#L14)):*
<details><summary><i>Previously, uniqueiniquity (Ben Lichtman) wrote…</i></summary><blockquote>

Is there a particular reason you need to wrap these types in a namespace?
As far as I can tell, it's not really helping you in any way, and you don't seem to be using it in your test file.
</blockquote></details>

There's no real reason. I was surprised by the fact that all top lever declaration should be exported, and I didn't want to add confusion to users with all those internal interfaces. What would you suggest would be better? Maybe I can move these to another file, like src/components/TwitterTimelineEmbed.ts, to reflect the JS lib.

**Comment by uniqueiniquity** [comment]
> There's no real reason. I was surprised by the fact that all top lever declaration should be exported, and I didn't want to add confusion to users with all those internal interfaces. What would you suggest would be better? Maybe I can move these to another file, like src/components/TwitterTimelineEmbed.ts, to reflect the JS lib.

I would recommend removing the namespace `timeline` and adding `export {};` to the bottom of the file. This will prevent those types from being exported.

File: types/react-twitter-embed/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+// Type definitions for react-twitter-embed 3.0
+// Project: https://github.com/saurabhnemade/react-twitter-embed#readme
+// Definitions by: Cyrille <https://github.com/zozoens31>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.5
```
Comment (uniqueiniquity) [inline types/react-twitter-embed/index.d.ts]: Is 3.5 really the minimum version of TypeScript required to compile these types? That seems overly restrictive.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Pull request closed without merging**